presto-main/src/main/java/com/facebook/presto/server/HttpRemoteTask.java

```
@@ -380,6 +394,23 @@ private synchronized void updateTaskInfo(TaskInfo newValue, List<TaskSource> sou
                 pendingSourceSplitCount -= removed;
             }
         }
+        int newSplitCount = getPartitionedSplitCount();
+
+        fireSplitChanged(newSplitCount - oldSplitCount);
+    }
+
+    private synchronized void fireSplitChanged(int delta)
+    {
+        if (delta != 0) {
+            for (SplitChangeListener splitChangeListener : splitChangeListeners) {
+                try {
+                    splitChangeListener.splitChanged(delta);
+                }
+                catch (Throwable e) {
+                    log.error(e, "Error notifying split change listener");
```

Swallowing exception does it sound to me as a good idea.